src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1767,11 +1865,15 @@ bool entitiesQuery
       // FIXME P4: Once issue #1705 is implemented, this check can be removed.
       if (qStringFilters(sco->value, filters) != true)
       {
-        if (badInputP)
+        if (badInputP)  // Bad Input reported by higher level
         {
           *badInputP = true;
           *err       = "invalid query expression";
         }
+        else  // There is a certain risk that this error is reported twice ...
```

Comment:
Well, every time we call alarmMgr.badInput, there is a risk that it is already called (either in a lower level, or in a higher level).
And, if it isn't called right now, a call could easily be added in the future.

This with maintaining only **one** badInput call per Bad Input is pretty difficult without any kind of tool for error-stacking (or whatever it is called).

Only that, nothing more.